admin_app/src/app/urgency-rules/UDSidebar.tsx

```diff
@@ -0,0 +1,219 @@
+  const toggleJsonModal = (): void => setJsonModalOpen(!jsonModalOpen);
+
+  if (loading) {
+    return <TypingAnimation />;
+  } else {
+    return (
+      <>
+        <Box display="flex" flexDirection="column" justifyContent="center">
+          {/* if type is string, there was an error */}
+          {typeof responseBoxData.parsedData === "string" ? (
+            <Typography variant="body1">
+              {responseBoxData.parsedData}
+            </Typography>
+          ) : (
+            <RenderUDResponse parsedData={responseBoxData.parsedData} />
+          )}
+          <Box
+            style={{
+              marginTop: "5px",
+              display: "flex",
+              justifyContent: "flex-end",
+              alignItems: "center",
+            }}
+          >
+            <Link
+              onClick={toggleJsonModal}
+              variant="caption"
+              align="right"
+              underline="hover"
+              sx={{ cursor: "pointer" }}
+            >
+              {"<json>"}
+            </Link>
+          </Box>
+        </Box>
+
+        <Modal
+          open={jsonModalOpen}
+          onClose={toggleJsonModal}
+          aria-labelledby="modal-modal-title"
+          aria-describedby="modal-modal-description"
+        >
+          <Fade in={jsonModalOpen}>
+            <Box
+              sx={{
+                position: "absolute",
+                top: "50%",
+                left: "50%",
+                transform: "translate(-50%, -50%)",
+                width: "80%",
+                maxHeight: "80%",
+                flexGrow: 1,
+                p: 4,
+                boxShadow: 24,
+                overflow: "scroll",
+                borderRadius: "10px",
+                bgcolor: "background.paper",
+              }}
+            >
+              <Box sx={{ display: "flex", justifyContent: "flex-end" }}>
+                <IconButton onClick={toggleJsonModal}>
+                  <Close />
+                </IconButton>
+              </Box>
+              <Typography
+                component={"span"}
+                id="modal-modal-description"
+                sx={{ marginTop: 2 }}
+              >
+                <pre
+                  style={{
+                    backgroundColor: "#f5f5f5",
+                    border: "1px solid #ccc",
+                    padding: "10px",
+                    borderRadius: "10px",
+                    overflowX: "auto",
+                    fontFamily: "Courier, monospace",
+                  }}
+                >
+                  {"rawJson" in responseBoxData
+                    ? JSON.stringify(responseBoxData.rawJson, null, 2)
+                    : "No JSON data found"}
+                </pre>
+              </Typography>
+            </Box>
+          </Fade>
+        </Modal>
+      </>
+    );
+  }
+};
+
+const UDSidebar = ({ closeSidebar }: { closeSidebar: () => void }) => {
```

Comment:
This doesn't extend all the way to the bottom for some reason. It seems to do so for the `SearchSidebar`. 

<img width="1573" alt="image" src="https://github.com/user-attachments/assets/37e60557-1533-4002-9e23-7e474a5d9f70">

Is this easy to fix?